internals/cli/cmd_run.go

```diff
@@ -202,13 +211,24 @@ func runDaemon(rcmd *cmdRun, ch chan os.Signal, ready chan<- func()) error {
 		} else {
 			logger.Noticef("Started default services with change %s.", changeID)
 		}
-	}
 
-	var stop chan struct{}
-	if ready != nil {
-		stop = make(chan struct{}, 1)
-		ready <- func() { close(stop) }
-		close(ready)
+		if ready != nil {
+			// wait for the default services to start
+			autoStartReady = make(chan struct{}, 1)
+			go func() {
+				waitCmd := waitMixin{
+					hideProgress: true,
+				}
+				waitCmd.setClient(rcmd.client)
+				_, err := waitCmd.wait(changeID)
+				if err != nil {
+					logger.Debugf("Error starting default services: %v", err)
```

Comment:
Should this be `logger.Noticef` instead of debug? Or even a hard error and exit with a non-zero error code?

Reply:
I have pushed a commit which produces a hard error if there is any error from `waitCmd.wait`. @woky mentioned this previously as well. 

Although looking at the [`internals/cli/wait.go`](https://github.com/canonical/pebble/blob/master/internals/cli/wait.go#L45-L152) file, I wonder if we should raise a hard-error **only if** we have received a non-`nil` "change" from the `wait()` function and `change.Err` is not `nil`. Let me know what you think.

Reply:
I agree. https://github.com/canonical/pebble/pull/257/commits/4098a71cd2ebe7125d54dd416e62554f3be8cfca should cover this case then.